### ISSUE TITLE
Build fails on branch-7-0 with -DLINK_STATIC_LIBMAPSERVER=1

### ISSUE THREAD

**Issue by dmorissette** [mapserver/mapserver#5170]
The changes in #5119 broke the build for branch-7-0 with -DLINK_STATIC_LIBMAPSERVER=1. We get the following error:

```
CMake Error at CMakeLists.txt:873 (set_target_properties):
  set_target_properties Can not find target to add properties to: mapserver
```
I was able to patch locally by adding a bunch of if (BUILD_DYNAMIC) and if(BUILD_STATIC) around the mapserver target calls to add the corresponding mapserver_static target calls, but that is probably not the right solution (I am a CMake illiterate, sorry).

I will let @sebastic test -DLINK_STATIC_LIBMAPSERVER=1 and provide a proper fix.

**Comment by sebastic** [comment]
On 2015-09-21 14:52, Daniel Morissette wrote:
> ```
> CMake Error at CMakeLists.txt:873 (set_target_properties):
>   set_target_properties Can not find target to add properties to: 
> mapserver
> ```

Does this issue still occur with the changes in the following commit?

https://github.com/mapserver/mapserver/commit/3c10ad17cbd04998edf779c92f3be04a9bc3b460

**Comment by dmorissette** [comment]
Yes, same problem. I was testing with a branch-7-0 checked out a few minutes ago.

**Comment by sebastic** [comment]
> I was able to patch locally by adding a bunch of if (BUILD_DYNAMIC) and if(BUILD_STATIC) around the mapserver target calls to add the corresponding mapserver_static target calls, but that is probably not the right solution (I am a CMake illiterate, sorry).

Since the CMake exports only make sense for the dynamically linked libmapserver using BUILD_{DYNAMIC,STATIC} seems appropriate. I only need two additional if statements though, see #5171.

**Comment by dmorissette** [comment]
Pull request #5171 solves my build problem. Note that I did not test the install steps, just the build steps, since my test environment is not setup to 'make install'.

**Comment by tbonfort** [comment]
erroneously closed with #5137
